File: spec/functional_test/fixtures/java/spring_callees/src/OrderController.java
package com.test;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/orders")
public class OrderController {

    @GetMapping("/legacy")
    public String legacy() {
        AuditLog.write("legacy");
        return getLegacy().toString();
    }

    private String getLegacy() {
        return "legacy";
    }
}

File: spec/functional_test/fixtures/kotlin/spring_callees/src/main/kotlin/com/example/OrderController.kt
package com.example

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/orders")
class OrderController {

    @GetMapping("/legacy")
    fun legacy(): String {
        AuditLog.write("legacy")
        return getLegacy().toString()
    }

    private fun getLegacy(): String {
        return "legacy"
    }
}
